inst/pages/exercises.qmd

```diff
@@ -1674,71 +1692,134 @@ p <- Heatmap(
+# Perform k-means clustering (k = 3) using the base R kmeans function
+set.seed(123)
+kmeans_result <- kmeans(mds_data, centers = 3)
 
-# Plot the reduced dimensions of pca_aitchison with clusters as colors
-plotReducedDim(pca_aitchison, colour_by = pca_aitchison_clusters)
-```
+# Add cluster labels to the SummarizedExperiment object
+# Convert clustering result to a factor
+tse$cluster <- factor(kmeans_result$cluster)
 
-Where: 
-- `reducedDim` is a function that extracts the reduced dimensionality
-representation of your data (e.g., from a PCA or PCoA object).
-- `plotReducedDim` is a function that can plot these reduced dimensions and
-has a 'colour_by argument' that accepts a vector of cluster assignments.
+# Visualize MDS with clusters
+plotReducedDim(tse, dimred = "MDS", colour_by = "cluster") +
+  labs(title = "MDS with k-means Clustering") +
+  theme_minimal()
+```
 
-#### 2. Heatmap with Annotations:
+3. Hierarchial Clustering
+- Hierarchical clustering of samples based on abundance data.
+- Visualizes the clustered data using a heatmap, with cluster information added to metadata.
 
-- Annotate the heatmap with additional metadata (e.g., subject groups or
-sample types).
+```{r}
+#| label: extra-hierarchialclustering-heatmap-ex3
+#| eval: FALSE
+#| code-fold: true
+#| code-summary: "Show solution"
 
-```{r, eval = FALSE}
-# Load necessary libraries
-library(SummarizedExperiment)
 library(ComplexHeatmap)
 
-# Example data creation
-# Assuming enterotype is a SummarizedExperiment object with sample metadata
-# and assay_data is a matrix containing assay data
+# Perform hierarchical clustering on the MDS data (extracted in last exercise)
+dist_matrix <- dist(mds_data)  # Compute distance matrix on MDS dimensions
+hclust_result <- hclust(dist_matrix, method = "complete")
+
+# Cut the dendrogram into 3 clusters
+cluster_labels <- cutree(hclust_result, k = 3)
+
+# Add cluster labels to the SummarizedExperiment object
+colData(tse)$cluster <- factor(cluster_labels)
+
+# Visualize MDS with clusters using scater::plotReducedDim
+plotReducedDim(tse, dimred = "MDS", colour_by = "cluster") +
+  labs(title = "MDS with Hierarchical Clustering") +
+  theme_minimal()
+
+# Optional: Visualize the hierarchical clustering with a heatmap
+# Convert MDS data to a matrix for heatmap visualization
+mds_matrix <- t(as.matrix(mds_data))  # Transpose the matrix to match the columns
+
+# Create a heatmap with cluster annotations
+heatmap <- Heatmap(mds_matrix,
+                   name = "MDS",
+                   show_row_dend = TRUE,
+                   show_column_dend = TRUE,
+                   column_split = cluster_labels,
+                   column_title = "Clusters",
+                   top_annotation = HeatmapAnnotation(cluster = cluster_labels),
+                   clustering_distance_columns = dist_matrix,
+                   clustering_method_columns = "complete")
+
+# Draw the heatmap
+draw(heatmap, heatmap_legend_side = "right")
```

Comment:
Perhaps you can get mds_matrix directly from mia::addCluster output? 

In general we support sechm for heatmaps because it supports the SE tool ecosystem, is there an advantage from using Heatmap.

Reply:
I had a talk with @ElySeraidarian, I appreciate his help. I am trying to do it with addCluster and sechm now.